runtime/trampolines.m

```diff
@@ -201,7 +201,17 @@
 					}
 				}
 			} else {
-				xamarin_assertion_message ("Don't know how to marshal a return value of type '%s.%s'. Please file a bug with a test case at https://github.com/xamarin/xamarin-macios/issues/new\n", mono_class_get_namespace (r_klass), mono_class_get_name (r_klass)); 
+				char *fullname = xamarin_class_get_full_name (r_klass, exception_gchandle);
+#if DOTNET
+				if (!strcmp (fullname, "System.IntPtr") || !strcmp (fullname, "ObjCRuntime.NativeHandle")) {
+#else
+				if (!strcmp (fullname, "System.IntPtr")) {
```

Comment:
There  are two functions that are much faster and simpler: `xamarin_is_class_intptr` and `xamarin_is_class_nativehandle`, no need to compute a full type name nor compare strings.